crates/isograph_lang_parser/src/parse_iso_literal.rs

```diff
@@ -19,11 +22,44 @@ use crate::{
     ParseResultWithLocation, ParseResultWithSpan, PeekableLexer,
 };
 
-pub fn parse_iso_fetch(
-    iso_fetch_text: &str,
+pub enum IsoLiteralExtractionResult {
+    ClientFieldDeclaration(WithSpan<ResolverDeclaration>),
+    EntrypointDeclaration(WithSpan<EntrypointTypeAndField>),
+}
+
+pub fn parse_iso_literal(
+    iso_literal_text: &str,
+    definition_file_path: ResolverDefinitionPath,
+    text_source: TextSource,
+) -> Result<WithSpan<IsoLiteralExtractionResult>, WithLocation<IsographLiteralParseError>> {
+    let mut tokens = PeekableLexer::new(iso_literal_text);
+    tokens
+        .with_span(|tokens| {
+            let discriminator: WithSpan<StringKey> = tokens
+                .parse_string_key_type(IsographLangTokenKind::Identifier)
+                .map_err(|with_span| with_span.map(IsographLiteralParseError::from))
+                .map_err(|err| err.to_with_location(text_source))?;
+            match discriminator.item.lookup() {
```

Comment:
let discriminator = tokens
                .parse_source_of_kind(IsographLangTokenKind::Identifier)
                .map_err(|with_span| with_span.map(IsographLiteralParseError::from))
                .map_err(|err| err.to_with_location(text_source))?;
            match tokens.source(discriminator.span) {

Reply:
What's the difference between parse_source_of_kind and parse_string_key_type?

Reply:
Fixed with [`99ee3eb` (#4)](https://github.com/isographlabs/isograph/pull/4/commits/99ee3eb52baeddc081badcdc8866d878dc4e97ad)

Reply:
`parse_source_of_kind` does not intern the returned string. I'll admit, the naming isn't helpful here.